tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -471,22 +474,36 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     HloInstruction *one = instr->AddInstruction(
         HloInstruction::CreateConstant(one_literal.Clone()));
 
-    // cuBLASLt FP8 GEMM kernels require A, which is later exchanged with B, to
-    // be transposed.
-    HloInstruction *b_transp =
-        instr->AddInstruction(HloInstruction::CreateTranspose(
-            ShapeUtil::MakeShape(
-                b->shape().element_type(),
-                {b->shape().dimensions(1), b->shape().dimensions(0)}),
-            b, {1, 0}));
+    // cuBLASLt FP8 GEMM kernels currently require the first operand, i.e. A, to
+    // be transposed. If the result of the GEMM is not in column major order, A
+    // and B are later exchanged, and B is transposed here instead.
+    // TODO(philipphack): Remove once cuBLASLt supports the NN configuration.
+    TF_ASSIGN_OR_RETURN(auto gemm_config,
+                        instr->backend_config<GemmBackendConfig>());
+    HloInstruction *a_or_b_transp;
+    TF_ASSIGN_OR_RETURN(bool is_col_major,
+                        OutputIsColumnMajor(instr, gemm_config));
+    if (is_col_major) {
+      a_or_b_transp = instr->AddInstruction(HloInstruction::CreateTranspose(
```

Comment:
nit: The logic would be clearer if you directly set the variable `a` to the transpose instruction here. And similarly for the `else` clause.